test/e2e/tests/eg_lifecycle.go

```diff
@@ -19,13 +19,138 @@ import (
+			ns := "gateway-lifecycle-infra"
+			routeNN := types.NamespacedName{
+				Name:      "http-backend-eg-un-install",
+				Namespace: ns,
+			}
+			gwNN := types.NamespacedName{
+				Name:      "lifecycle-gateway",
+				Namespace: ns,
+			}
+			gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig,
+				suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/eg-uninstall-install",
+				},
+				Response: http.Response{
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+			req := http.MakeRequest(t, &expectedResponse, gwAddr, "HTTP", "http")
+			cReq, cResp, err := suite.RoundTripper.CaptureRoundTrip(req)
+			if err != nil {
+				t.Errorf("failed to get expected response before the install process started: %v", err)
+			}
+			if err := http.CompareRequest(t, &req, cReq, cResp, expectedResponse); err != nil {
+				t.Errorf("failed to compare request and response before the uninstall and install process started: %v", err)
+			}
+
+			// we will start the uninstallation process after ensuring that the envoy gateway normal service.
+			relName := "eg"
+			relNamespace := "envoy-gateway-system"
+			options.DefaultConfigFlags.Namespace = ptr.To(relNamespace)
+
+			ht := helm.NewPackageTool()
+			if err = ht.Setup(); err != nil {
+				t.Errorf("failed to setup of packageTool: %v", err)
+			}
+
+			t.Log("start uninstall envoy gateway resources")
+			if err := ht.RunUninstall(&helm.PackageOptions{
+				ReleaseName: relName,
+			}); err != nil {
+				t.Errorf("failed to uninstall envoy-gateway: %v", err)
+			}
+			t.Log("success to uninstall envoy gateway resources")
+		})
+		t.Run("Install Envoy Gateway should succeed", func(t *testing.T) {
+			ns := "gateway-lifecycle-infra"
+			routeNN := types.NamespacedName{
+				Name:      "http-backend-eg-un-install",
+				Namespace: ns,
+			}
+			gwNN := types.NamespacedName{
+				Name:      "lifecycle-gateway",
+				Namespace: ns,
+			}
+			gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig,
+				suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/eg-uninstall-install",
+				},
+				Response: http.Response{
+					StatusCode: 200,
+				},
+				Namespace: ns,
+			}
+			req := http.MakeRequest(t, &expectedResponse, gwAddr, "HTTP", "http")
+
+			relName := "eg"
+			relNamespace := "envoy-gateway-system"
+			options.DefaultConfigFlags.Namespace = ptr.To(relNamespace)
+			lastVersionTag := os.Getenv("last_version_tag")
+			if lastVersionTag == "" {
+				lastVersionTag = "v0.0.0-latest" // Default version tag if not specified
+			}
+
+			ht := helm.NewPackageTool()
+			if err := ht.Setup(); err != nil {
+				t.Errorf("failed to setup of packageTool: %v", err)
+			}
+
+			t.Log("start install envoy gateway resource")
```

Comment:
I think that there's also some value if the installation will install the local version in the branch (from source), instead of v0.0.0-latest. This way, we're validating the contents of the PR/CI branch, instead of the installability of a previously-released artifact. Plus, in its current state, the test may impact other tests in this suite, by changing the actual EG version deployed (from local branch to v0.0.0-latest). 

@arkodg, @ShyunnY  - WDYT? Should we install from sources or atleast cleanup at the end of the test and revert to the local branch version?

Reply:
@guydc  
This is a good suggestion, but I don't think it's necessary. We are focusing on `helmtool` conformance testing, and I don't think we should add complexity by verifying PR/CI here.

In addition, we specify the order of execution in the Makefile, will this still affect other test suites?